src/htmlhelp.cpp

```diff
@@ -134,7 +134,17 @@ static QCString field2URL(const IndexField *f,bool checkReversed)
   QCString result = f->url + Doxygen::htmlFileExtension;
   if (!f->anchor.isEmpty() && (!checkReversed || f->reversed)) 
   {
-    result+="#"+f->anchor;  
+    result+="#";
```

Comment:
I would expect this could be written much shorter using substitute,e.g.: `result+="#"+substitute(f->anchor,":","%3A");`